src/lib.rs

```diff
@@ -42,6 +43,43 @@ pub use dataset::{
 pub use driver::Driver;
 pub use metadata::Metadata;
 
+/// Apply GeoTransform to x/y coordinate.
+/// Wraps [GDALApplyGeoTransform].
+///
+/// [GDALApplyGeoTransform]: https://gdal.org/api/raster_c_api.html#_CPPv421GDALApplyGeoTransformPdddPdPd
+pub fn apply_geo_transform(geo_transform: &GeoTransform, pixel: f64, line: f64) -> (f64, f64) {
```

Comment:
Would it be more ergonomic to throw these inside a trait that is impl by `GeoTransform` so we get the nicer:

```rust
let (wx, wy) = geo_t.apply(pix, line);
let inv_t = geo_t.invert()?;
```

Optional suggestion.